io/pci/distro_tools.py

```diff
@@ -30,27 +30,61 @@ class DisrtoTool(Test):
     '''
     to test different type of tool
     '''
-    @skipUnless(IS_POWER_VM,
-                "supported only on PowerVM platform")
+
     def setUp(self):
         '''
-        to check host platform
+        get all parameters
         '''
-        self.option = self.params.get("option", default='')
+        self.option = self.params.get("test_opt", default='').split(",")
+        self.tool = self.params.get("tool", default='')
 
-    def test(self):
+    @skipUnless(IS_POWER_VM,
+                "supported only on PowerVM platform")
+    def lsslot(self, option):
         '''
-        test the lsslot tool
+        run lsslot
         '''
         cmd = "lsslot"
-        if self.option == "pci":
-            cmd = "%s -d %s" % (cmd, self.option)
+        if option == "pci":
+            cmd = "%s -d %s" % (cmd, option)
```

Comment:
is this '-c'?
your trying to display the hotplugable slots which can be achieved by using "lsslot -c pci"

Usage: lsslot [-c | -a | -b | -p | -o | -s ][-F | -d | -w]
        -c <connector type>
                Display the slots of the specified connector type.  The valid
                connector types are "pci" for hotplug PCI slots, "slot" for
                logical slots, "phb" for PHB's, "port" for LHEA ports, "mem"
                for memory, and "cpu" for cpu's. The default
                is "slot" if no -c option is specified.

Reply:
so i will removed -d pci option ..?

Reply:
check what is the use case of option 'd' and then go accordingly.
I feel option 'd' can be used for virtual slots and not for physical slot.
So if we plan to add virtual slot verification then using option 'd' could be fine.